lib/reek/smells/uncommunicative_parameter_name.rb

```diff
@@ -45,14 +40,15 @@ def self.default_config
       #
       # @return [Array<SmellWarning>]
       #
-      def examine_context(ctx)
-        context_expression = ctx.exp
-        context_expression.parameter_names.select do |name|
-          bad_name?(ctx, name) && ctx.uses_param?(name)
+      def examine_context(context)
+        expression = context.exp
+        expression.parameter_names.select do |name|
+          sanitized_name = sanitize name
+          !acceptable_name?(context: context, name: sanitized_name) && context.uses_param?(name)
```

Comment:
Hmmm, I agree that it's a tad complex but at the same time I find this is very readable. "If name is not acceptable and the param is used -> smell".

I could move

```Ruby
!acceptable_name?(context: context, name: sanitized_name) && context.uses_param?(name)
```

to something like:

```Ruby
def uncommunicative_parameter_name(name: sanitized_name, context: context)
  !acceptable_name?(context: context, name: sanitized_name) && context.uses_param?(name)
end
```

What do you think about that?